### PR TITLE
Migrating various Firebolt APIs into L-SDK

### PR DESCRIPTION
**Pull request by jlacivita**

### CODE REVIEW

**Comment by bengreenberg** [comment]
I think It makes sense to have XVP stuff segmented, not sure about Ad Framework but maybe.

I dont think the name is right - there is Firebolt stuff elsewhere in the SDK as well so that seems confusing.

**Comment by jlacivita** [comment]
Okay, I moved Advertising and Personalization back into Platform via a Profile submodule.

Also took a better (I think) attempt at matching L-SDK style w/ methods that have longer names.

File: src/Discovery/index.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import Log from '../Log'
+
+let getEntitlements = function() {
+  return {
```
Comment (bengreenberg) [inline src/Discovery/index.js:23]: still think this should not return values - we don't have a way of getting them from XVP.

File: src/Discovery/index.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+let setEntitlements = function() {
+  return true;
+}
+
+let watched = function(watchedItems) {
+  watchedItems.reduce( (items, item) => { Log.info('Added ' + watchedItems.watchedOn + ": " + watchedItems.contentId )})
+  return true
+}
+
+let dashBoardTile = function(name, imageUrl, linkUrl) {
```
Comment (bengreenberg) [inline src/Discovery/index.js:44]: this is missing a bunch of parameters, probably should take an object. see https://ottx-docs.wc-g1.cf.comcast.net/docs/app-launchpad-tiles/

Comment (bengreenberg) [reply src/Discovery/index.js:44]: also we need a better name for this....

File: src/Ads/index.js [hunk 3]
```diff
@@ -16,223 +16,77 @@
...
+  return '26ccd5a7b2c2a50e7d4b2244e9d4c048'
 }
 
-const playSlot = (slot = []) => {
-  return slot.reduce((promise, ad) => {
-    return promise.then(() => {
-      return playAd(ad)
-    })
-  }, Promise.resolve(null))
+let clearAdvertisingId = function() {
```
Comment (bengreenberg) [inline src/Ads/index.js:32]: I think this is better called `resetAdvertisingId` as you still have an ad id at the end of it, just a new one.

File: src/Ads/index.js [hunk 4]
```diff
@@ -16,223 +16,77 @@
...
-        cleanup()
-      },
-      // todo: pause, resume, mute, unmute beacons
-    }
-    // add all listeners
-    Object.keys(handlers).forEach(handler => videoEl.addEventListener(handler, handlers[handler]))
-
-    videoEl.play()
-  })
+let getConfig = function() {
```
Comment (bengreenberg) [inline src/Ads/index.js:36]: We may want to make this `getAdPlatformConfig` or something, as its to expected to be used generically,.

File: src/Ads/index.js [hunk 5]
```diff
@@ -16,223 +16,77 @@
...
+    adSkipTier: 'NOSKIP_NORMAL_SPEED',		
+    adSkipGracePeriodSeconds: 60		
   }
 }
 
-const fireOnConsumer = (event, args) => {
-  if (consumer) {
-    consumer.fire('$ad' + event, args)
-    consumer.fire('$adEvent', event, args)
+let getPrivacy = function() {
```
Comment (bengreenberg) [inline src/Ads/index.js:50]: we need to work on the output format here, it needs to comply with some industry standards as badger does, I think.

File: src/Ads/index.js [hunk 6]
```diff
@@ -16,223 +16,77 @@
...
 export const initAds = config => {
   if (config.getAds) {
-    getAds = config.getAds
+    initAdsPlayer(config)
   }
 }
 
-const state = {
-  active: false,
+let getAdvertisingId = function() {
```
Comment (bengreenberg) [inline src/Ads/index.js:28]: we also need to add methods:
`deviceAdAttributes()` (returns object)
`appStoreId()` (returns string)

File: src/Discovery/index.js (hunk 1 above)
Comment (jlacivita) [reply src/Discovery/index.js:23]: Well that sort of ends that discussion :) I'll remove

File: src/Discovery/index.js (hunk 2 above)
Comment (jlacivita) [reply src/Discovery/index.js:44]: Okay... renaming `watchNext` as discussed. will add the missing params

File: src/Ads/index.js (hunk 3 above)
Comment (jlacivita) [reply src/Ads/index.js:32]: oh, good feedback

**Pull request closed without merging**